cpp/order/Steinhardt.cc

```diff
@@ -0,0 +1,391 @@
+            iter != sph_eval.end(); ++iter)
+            {
+            Ylm[j] = *iter;
+            ++j;
+            }
+        }
+    }
+
+template <typename T>
+std::shared_ptr<T> Steinhardt::makeArray(size_t size)
+    {
+    return std::shared_ptr<T>(new T[size], std::default_delete<T[]>());
+    }
+
+void Steinhardt::reallocateArrays(unsigned int Np)
+    {
+    m_Qlmi = Steinhardt::makeArray<complex<float> >((2*m_l+1) * Np);
+    m_Qli = Steinhardt::makeArray<float>(Np);
+    m_Qlm = Steinhardt::makeArray<complex<float> >(2*m_l+1);
+
+    if (m_average)
+        {
+        m_QlmiAve = Steinhardt::makeArray<complex<float> >((2*m_l+1) * Np);
+        m_QliAve = Steinhardt::makeArray<float>(Np);
+        }
+
+    if (m_Wl)
+        {
+        m_WliOrder = Steinhardt::makeArray<complex<float> >(Np);
+        }
+    }
+
+void Steinhardt::compute(const box::Box& box,
+						 const locality::NeighborList *nlist,
+						 const vec3<float> *points,
+						 unsigned int Np)
+    {
+    // Conditionally reinitialize arrays if size differs from previous call.
+    if (m_Np != Np)
+        {
+        m_Np = Np;
+        Steinhardt::reallocateArrays(Np);
+        }
+
+    // Computes the base Qlmi required for each specialized order parameter
+    Steinhardt::baseCompute(box, nlist, points);
+
+    if (m_average)
+        {
+        Steinhardt::computeAve(box, nlist, points);
+        }
+
+	// Reduce Qlm
+	Steinhardt::reduce();
+
+    if (m_Wl)
+        {
+        memset((void*) m_WliOrder.get(), 0, sizeof(complex<float>)*m_Np);
+        if (m_average)
+            {
+            Steinhardt::aggregateWl(m_WliOrder, m_QlmiAve);
+			m_NormWl = Steinhardt::normalizeWl();
+            }
+        else
+            {
+            Steinhardt::aggregateWl(m_WliOrder, m_Qlmi);
+			m_NormWl = Steinhardt::normalizeWl();
+            }
+        }
+        {
+        if (m_average)
+            {
+            m_Norm = Steinhardt::normalize();
+            }
+        else
+            {
+            m_Norm = Steinhardt::normalize();
+            }
+        }
+    }
+
+void Steinhardt::baseCompute(const box::Box& box,
+							 const locality::NeighborList *nlist,
+							 const vec3<float> *points)
+    {
+    nlist->validate(m_Np, m_Np);
+
+    memset((void*) m_Qlmi.get(), 0, sizeof(complex<float>)*(2*m_l+1)*m_Np);
+    memset((void*) m_Qli.get(), 0, sizeof(float)*m_Np);
+
+    parallel_for(tbb::blocked_range<size_t>(0, m_Np),
+        [=] (const blocked_range<size_t>& r)
+        {
+        const float rminsq = m_rmin * m_rmin;
+        const float rmaxsq = m_rmax * m_rmax;
+        const float normalizationfactor = 4*M_PI/(2*m_l+1);
+        const size_t *neighbor_list(nlist->getNeighbors());
+
+		if (!m_average)
+			{
+			bool Qlm_exists;
+			m_Qlm_local.local(Qlm_exists);
+			if (!Qlm_exists)
+				{
+				m_Qlm_local.local() = new complex<float> [2*m_l+1];
+				memset((void*) m_Qlm_local.local(), 0, sizeof(complex<float>) * (2*m_l+1));
+				}
+			}
```

Comment:
Fix the indentation of this block, and I think you can also get rid of the memset since `reallocateArrays` does it.